sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -118,6 +118,19 @@ class AttributeMap : public std::unordered_map<std::string, OwnedAttributeValue>
         });
   }
 
+  // Construct attribute map and populate with optional attributes
+  AttributeMap(const opentelemetry::common::KeyValueIterable *attributes) : AttributeMap()
+  {
+    if (attributes != nullptr)
+    {
+      attributes->ForEachKeyValue(
+          [&](nostd::string_view key, opentelemetry::common::AttributeValue value) noexcept {
+            SetAttribute(key, value);
+            return true;
```

Comment:
nit - some code duplication, which can be moved to a separate utils method. But don't see it really required.